styx-cli/src/main/java/com/spotify/styx/cli/Main.java

```diff
@@ -379,16 +379,21 @@ private void workflowShow() throws ExecutionException, InterruptedException, IOE
     final String wid = namespace.getString(parser.workflowShowWorkflowId.getDest());
 
     // Fetch config and state in parallel
-    final CompletableFuture<ByteString> workflowResponse =
+    final CompletableFuture<Response<ByteString>> workflowResponse =
         sendAsync(Request.forUri(apiUrl("workflows", cid, wid))).toCompletableFuture();
-    final CompletableFuture<ByteString> stateResponse =
+    final CompletableFuture<Response<ByteString>> stateResponse =
         sendAsync(Request.forUri(apiUrl("workflows", cid, wid, "state"))).toCompletableFuture();
 
     // Wait for both responses
     CompletableFuture.allOf(workflowResponse, stateResponse).get();
 
-    final Workflow workflow = deserialize(workflowResponse.get(), Workflow.class);
-    final WorkflowState workflowState = deserialize(stateResponse.get(), WorkflowState.class);
+    final ByteString workflowPayload = requireSuccess(workflowResponse.get())
+        .payload().orElse(ByteString.EMPTY);
+    final ByteString statePayload = requireSuccess(stateResponse.get())
+        .payload().orElse(ByteString.EMPTY);
+
+    final Workflow workflow = deserialize(workflowPayload, Workflow.class);
```

Comment:
I guess the `AutoValue_Workflow` will throw NPE. That's true for other methods like `resourceShow` as well. The cli doesn't really seem to validate anything it gets from the styx api.